wordpress-mobile/WordPress-iOS
# Comment: convert `PostContentProvider` protocol methods to Swift

Ref: #16876

This converts `Comment`'s `PostContentProvider` protocol methods to Swift.

Notes: 
- This is the first step in converting Comment to Swift, so I've started with Swift extensions. In the end, `Comment.swift` will be _the_ Comment.
- I didn't see a need for `WPCommentContentViewProvider` as it was only used by `Comment`. So I removed that and used `PostContentProvider` directly.
- `blogNameForDisplay` and `statusForDisplay` don't appear to be used (or I can't find where). I've simply converted the code for now and will follow up on this later.

To test:

Go to these views and verify these components are displayed as expected. That is, there should be no visible change.

Site > Comments list:
- Author name
- Author avatar
- Comment title
- Comment content

Site > Comments > Comment details:
- Author name
- Author avatar
- Comment title
- Comment content

Reader > post comments:
- Author name
- Author avatar
- Author URL (the author's name is linked to this)
- Comment creation date


## Regression Notes
1. Potential unintended areas of impact
The listed Comment components could be displayed incorrectly.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Compared each component with how it was before.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
